File: pytools/examples/golang/src/client/proto/msg_scene_rot_pos.go
package proto

import (
	"packet"
)

type MsgSceneRotPos struct {
	rotX                     int16
	rotY                     int16
	rotZ                     int16
	posX                     int16
	posY                     int16
	posZ                     int16
}

func (this *MsgSceneRotPos) Encode() []byte {
	pack := packet.NewWriteBuff(64)

	pack.WriteInt16(this.rotX)
	pack.WriteInt16(this.rotY)
	pack.WriteInt16(this.rotZ)
	pack.WriteInt16(this.posX)
	pack.WriteInt16(this.posY)
	pack.WriteInt16(this.posZ)

	return pack.ReadBytes()
}

func MsgSceneRotPosDecode(pack *packet.Packet) *MsgSceneRotPos {
	msgSceneRotPos := &MsgSceneRotPos{}

	msgSceneRotPos.rotX = pack.ReadInt16()
	msgSceneRotPos.rotY = pack.ReadInt16()
	msgSceneRotPos.rotZ = pack.ReadInt16()
	msgSceneRotPos.posX = pack.ReadInt16()
	msgSceneRotPos.posY = pack.ReadInt16()
	msgSceneRotPos.posZ = pack.ReadInt16()
	return msgSceneRotPos
}

func (this *MsgSceneRotPos) SetRotX(rotX int16) {
	this.rotX = rotX
}

func (this *MsgSceneRotPos) GetRotX() int16 {
	return this.rotX
}

func (this *MsgSceneRotPos) SetRotY(rotY int16) {
	this.rotY = rotY
}

func (this *MsgSceneRotPos) GetRotY() int16 {
	return this.rotY
}

func (this *MsgSceneRotPos) SetRotZ(rotZ int16) {
	this.rotZ = rotZ
}

func (this *MsgSceneRotPos) GetRotZ() int16 {
	return this.rotZ
}

func (this *MsgSceneRotPos) SetPosX(posX int16) {
	this.posX = posX
}

func (this *MsgSceneRotPos) GetPosX() int16 {
	return this.posX
}

func (this *MsgSceneRotPos) SetPosY(posY int16) {
	this.posY = posY
}

func (this *MsgSceneRotPos) GetPosY() int16 {
	return this.posY
}

func (this *MsgSceneRotPos) SetPosZ(posZ int16) {
	this.posZ = posZ
}

func (this *MsgSceneRotPos) GetPosZ() int16 {
	return this.posZ
}

File: pytools/data/data.code.cpp.server/MsgSceneRotPos.cpp
#ifndef _MSG_SCENE_ROT_POS_
#define _MSG_SCENE_ROT_POS_

#include <list>

#include <pb_type.hpp>
#include <Packet.hpp>

#include <Msg.hpp>



class MsgSceneRotPos
{
private:
	I16 m_rot_x;
	I16 m_rot_y;
	I16 m_rot_z;
	I16 m_pos_x;
	I16 m_pos_y;
	I16 m_pos_z;


public:
	BYTE* Encode(U32* len)
	{
		Packet packet;
		packet.WriteI16(m_rot_x);
		packet.WriteI16(m_rot_y);
		packet.WriteI16(m_rot_z);
		packet.WriteI16(m_pos_x);
		packet.WriteI16(m_pos_y);
		packet.WriteI16(m_pos_z);
		return packet.ReadBytes(len);
	}

	MsgSceneRotPos()
	{

	}

	MsgSceneRotPos(Packet* packet)
	{
		m_rot_x = packet->ReadI16();
		m_rot_y = packet->ReadI16();
		m_rot_z = packet->ReadI16();
		m_pos_x = packet->ReadI16();
		m_pos_y = packet->ReadI16();
		m_pos_z = packet->ReadI16();
	}

	BYTE* GetBytes(U32* len)
	{
		return Encode(len);
	}


	void SetRotX(I16 rot_x)
	{
		m_rot_x = rot_x;
	}
	I16 GetRotX()
	{
		return m_rot_x;
	}

	void SetRotY(I16 rot_y)
	{
		m_rot_y = rot_y;
	}
	I16 GetRotY()
	{
		return m_rot_y;
	}

	void SetRotZ(I16 rot_z)
	{
		m_rot_z = rot_z;
	}
	I16 GetRotZ()
	{
		return m_rot_z;
	}

	void SetPosX(I16 pos_x)
	{
		m_pos_x = pos_x;
	}
	I16 GetPosX()
	{
		return m_pos_x;
	}

	void SetPosY(I16 pos_y)
	{
		m_pos_y = pos_y;
	}
	I16 GetPosY()
	{
		return m_pos_y;
	}

	void SetPosZ(I16 pos_z)
	{
		m_pos_z = pos_z;
	}
	I16 GetPosZ()
	{
		return m_pos_z;
	}

};

#endif
